controllers/mirrorpeer_controller.go

```diff
@@ -508,3 +518,80 @@ func (r *MirrorPeerReconciler) createDRClusters(ctx context.Context, mp *multicl
 	}
 	return nil
 }
+
+func (r *MirrorPeerReconciler) createRoleBindingsForSpoke(ctx context.Context, peer multiclusterv1alpha1.MirrorPeer) error {
+	crb := rbacv1.ClusterRoleBinding{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: spokeClusterRoleBindingName}, &crb)
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+	}
+	var subjects []rbacv1.Subject
+	if crb.Subjects != nil {
+		subjects = crb.Subjects
+	}
+
+	// Add users
+	for _, pr := range peer.Spec.Items {
+		sub := getSubjectByPeerRef(pr, "User")
+		if len(subjects) == 0 {
+			subjects = append(subjects, *sub)
+		} else {
+			if !utils.ContainsSubject(subjects, sub) {
+				subjects = append(subjects, *sub)
+			}
+		}
```

Comment:
you dont need len(subjects) == 0 check, else part will takecare of appending if the subject is empty or not found